libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -72,6 +74,18 @@ fun createNavigationRoute(
     directionsRoute
         .toNavigationRoute(sdkRouteParser, com.mapbox.navigation.base.route.RouterOrigin.Custom())
 
+/**
+ * Internal API used for testing purposes. Needed to avoid calling native parser from unit tests.
+ */
+@TestOnly
+fun createNavigationRoutes(
```

Comment:
I'm trying to avoid static mocking in general.

1. It's extremely slow. Here's an example of how absence of static mocking saved 3 minutes on CI https://github.com/mapbox/mapbox-navigation-android/pull/5789
2. For me it's an indicator of code smell. If you can't test without static mocking, it's something wrong with your architecture. Good code is always easy to test. I'm always trying to iterate until testing become simple.
3. I needed navigation routes in every test. It's much easier to create a function for this rather than do static mocking every time.

Reply:
Good question! I need some time to think about it 🙂

Reply:
also, you provided an example with Logger frontend was injected into almost every class, which really affects CI build time. This one is just used for a limited amount of cases, that shouldn't affect so much

Reply:
> > For me it's an indicator of code smell. If you can't test without static mocking, it's something wrong with your architecture.
> 
> But you've just introduced a test-only method that just proxies all the arguments to the static method. What's the difference?

The difference is that I can't use super power of static mocking to start testing implementation details. Tests which don't touch implementation details are less fragile. I value it a lot. 

But I mush admit that for this specific case reasoning in an order of importance is: 3, 1, 2

Reply:
But in order to use the real method you must pass a valid json, right? And the json required to create a navigation route is pretty complicated. Isn't is easier to just mock the fields in the route that interest you?
Nothing blocking tho.